File: app/src/main/java/com/example/vinheira_agnello_f4/data/Produto.kt
package com.example.vinheira_agnello_f4.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "produtos")
data class Produto(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val nome: String,
    val descricao: String? = null,
    val uva: String? = null,
    val ano: Int? = null,
    val preco: Double,
    val quantidade: Int
)

File: backend/EstoqueVinheria/Produto.cs
using System.ComponentModel.DataAnnotations;

public class Produto
{
    [Key]
    public int Id { get; set; }

    [Required]
    public string Nome { get; set; } = "";

    public string? Descricao { get; set; }

    [Required]
    public double Preco { get; set; }

    [Required]
    public int Quantidade { get; set; }
}
